src/hex.rs

```diff
@@ -609,3 +609,38 @@ fn concat_test() -> Result<()> {
     assert_eq!(a.concat(b), Hex::from_str("DE-AD-BE-EF")?);
     Ok(())
 }
+
+#[test]
+fn creates_from_big_slice() -> Result<()> {
+    let s: [u8; 9] = [0xAB, 0xD8, 0xAB, 0xD8, 0xAB, 0xD8, 0xAB, 0xD8, 0xAB];
+    let mut accum = vec![];
+    for el in s {
+        accum.push(el);
+        accum.push(el);
+        accum.push(el);
+    }
+    let h = Hex::from_slice(accum.as_slice());
+    assert_eq!(27, h.len());
+    assert_eq!(h.to_vec(), accum);
+    Ok(())
+}
+
+#[test]
+fn concatenates_from_hex_vec() -> Result<()> {
+    let a = Hex::from_vec(vec![0x12, 0xAB]);
+    let b = Hex::from_slice("as_bytesss".as_bytes());
+    let c = Hex::from_vec(vec![0x12, 0xAD]);
+    let res = a.concat(b).concat(c);
+    assert!(!res.is_empty());
```

Comment:
@l3r8yJ maybe you can also check the length of the result here? Just checking for non-emptiness seems to be too weak.